src/components/main/Main.js

```diff
@@ -0,0 +1,47 @@
+import "./Main.css";
+
+import Card from "../card/Card";
+import { useState, useEffect } from "react";
+
+function Main() {
+  const [error, setError] = useState(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [characters, setCharacters] = useState([]);
+
+  useEffect(() => {
+    fetch("http://hp-api.herokuapp.com/api/characters").then((res) =>
+      res.json().then(
+        (result) => {
+          setIsLoaded(true);
+          setCharacters(result);
+        },
+        (error) => {
+          setIsLoaded(true);
+          setError(error);
+        }
+      )
+    );
+  }, []);
+  if (error) {
+    return <div>Error : {error.message}</div>;
+  } else if (!isLoaded) {
+    return <div>Loading...</div>;
+  } else {
+    return (
+      <main className="main">
+        {characters.map((character) => (
+          <Card
+            key={character.name}
+            name={character.name}
+            img={character.image}
+            gender={character.gender}
+            dateOfBirth={character.dateOfBirth}
+            house={character.house}
+          />
+        ))}
+      </main>
+    );
+  }
```

Comment:
Could be done in a different way:
```suggestion
    return (
      <main className="main">
        {error && <div>Error : {error.message}</div>}
        {characters ? (characters.map((character) => (
          <Card
            key={character.name}
            name={character.name}
            img={character.image}
            gender={character.gender}
            dateOfBirth={character.dateOfBirth}
            house={character.house}
          />
        ))) :
        (<div>Loading...</div>)
        }
      </main>
    );
  }
```
This way you don't need the `isLoaded` state and take away a bit of the complexity of the return statement.

Reply:
Looks way better. I actually looked up ways to fetch data from an API and followed the example from the react.js documentation. Thought it was considered best practice to take care of the loading state as well.

Reply:
It's pretty cool that you looked it up and from what I see is that it is working the intended way, right? 🙂 I am not 100% sure what would be considered best practice here, I just wanted to show a different approach. The way I learned it is, that the less state we need to handle the better. But I guess at some point this is also personal preference. Hope this helps?